autoload/vim_git_backup/git_helper.vim

```diff
@@ -115,7 +115,7 @@ endfunction
 "         The shell commands needed to generate a git commit message.
 "
 function! vim_git_backup#git_helper#get_commit_commands(root, file)
-    let l:file_name = fnamemodify(a:file, ':t')
+    let l:file_name = fnamemodify(a:file, ':.')
```

Comment:
Instead of the base name of the file, this uses the file name from the current directory so files with the same name in different paths are differentiated.